python/cuml/test/test_pickle.py

```diff
@@ -0,0 +1,201 @@
+)
+
+solver_models = dict(
+    CD=cuml.CD(),
+    SGD=cuml.SGD(eta0=0.005)
+)
+
+cluster_models = dict(
+    KMeans=cuml.KMeans()
+)
+
+decomposition_models = dict(
+    PCA=cuml.PCA(),
+    TruncatedSVD=cuml.TruncatedSVD(),
+    UMAP=cuml.UMAP()
+)
+
+neighbor_models = dict(
+    NearestNeighbors=cuml.NearestNeighbors()
+)
+
+
+def unit_param(*args, **kwargs):
+    return pytest.param(*args, **kwargs, marks=pytest.mark.unit)
+
+
+def quality_param(*args, **kwargs):
+    return pytest.param(*args, **kwargs, marks=pytest.mark.quality)
+
+
+def stress_param(*args, **kwargs):
+    return pytest.param(*args, **kwargs, marks=pytest.mark.stress)
+
+
+def make_cudf_series(arr):
+    df = pd.DataFrame(
+                           {'fea0': arr[:, ]})
+    df = df.values
+    df = df[:, 0]
+    return cudf.Series(df)
+
+
+def pickle_save_load(model):
+    os.mkdir('tmp')
+
+    pickle_file = 'tmp/cu_model'
+    try:
+        with open(pickle_file, 'wb') as pf:
+            pickle.dump(model, pf)
+    except (TypeError, ValueError) as e:
+        pf.close()
+        os.remove(pickle_file)
+        os.rmdir('tmp')
+        pytest.fail(e)
+
+    with open(pickle_file, 'rb') as pf:
+        cu_after_pickle_model = pickle.load(pf)
+
+    os.remove(pickle_file)
+    os.rmdir('tmp')
+
+    return cu_after_pickle_model
+
+
+def make_dataset(datatype, input_type, nrows, ncols):
+    train_rows = np.int32(nrows*0.8)
+    X, y = make_regression(n_samples=(nrows), n_features=ncols,
+                           random_state=0)
+    X_test = np.asarray(X[train_rows:, :]).astype(datatype)
+    X_train = np.asarray(X[:train_rows, :]).astype(datatype)
+    y_train = np.asarray(y[:train_rows, ]).astype(datatype)
+
+    if input_type == 'dataframe':
+        X_train = np_to_cudf(X_train)
+        y_train = make_cudf_series(y_train)
+        X_test = np_to_cudf(X_test)
+
+    return X_train, y_train, X_test
+
+
+@pytest.mark.parametrize('datatype', [np.float32, np.float64])
+@pytest.mark.parametrize('input_type', ['dataframe', 'ndarray'])
+@pytest.mark.parametrize('model', regression_models.values())
+@pytest.mark.parametrize('nrows', [unit_param(20)])
+@pytest.mark.parametrize('ncols', [unit_param(3)])
+def test_regressor_pickle(datatype, input_type, model, nrows, ncols,):
+    X_train, y_train, X_test = make_dataset(datatype, input_type, nrows, ncols)
+
+    model.fit(X_train, y_train)
+    cu_before_pickle_predict = model.predict(X_test).to_array()
+
+    cu_after_pickle_model = pickle_save_load(model)
+
+    cu_after_pickle_predict = cu_after_pickle_model.predict(X_test).to_array()
+
+    assert array_equal(cu_before_pickle_predict, cu_after_pickle_predict)
+
+
+@pytest.mark.parametrize('datatype', [np.float32, np.float64])
+@pytest.mark.parametrize('input_type', ['dataframe', 'ndarray'])
+@pytest.mark.parametrize('model', solver_models.values())
+@pytest.mark.parametrize('nrows', [unit_param(20)])
+@pytest.mark.parametrize('ncols', [unit_param(3)])

```

Comment:
regression_models and solver models tests could be combined into one since the APIs are identical. And really these solvers are all regression models by default...